contracts/REXTwoWayUniswapMarket.sol

```diff
@@ -0,0 +1,614 @@
+        addOutputPool(
+            _subsidyToken,
+            0,
+            _emissionRate,
+            77,
+            market.outputPools[OUTPUTA_INDEX].shareScaler
+        );
+        lastDistributionTokenAAt = block.timestamp;
+        lastDistributionTokenBAt = block.timestamp;
+        // Does not need to add subsidy token to outputPoolIndicies
+        // since these pools are hardcoded
+    }
+
+    function addOutputPool(
+        ISuperToken _token,
+        uint128 _feeRate,
+        uint256 _emissionRate,
+        uint256 _requestId,
+        uint128 _shareScaler
+    ) public override onlyOwner {
+        // Only Allow 4 output pools, this overrides the block in REXMarket
+        // where there can't be two output pools of the same token
+        require(market.numOutputPools < 4, "too many pools");
+
+        OutputPool memory _newPool = OutputPool(
+            _token,
+            _feeRate,
+            _emissionRate,
+            _shareScaler
+        );
+        market.outputPools[market.numOutputPools] = _newPool;
+        market.outputPoolIndicies[_token] = market.numOutputPools;
+        _createIndex(market.numOutputPools, _token);
+        market.numOutputPools++;
+        OracleInfo memory _newOracle = OracleInfo(_requestId, 0, 0);
+        market.oracles[_token] = _newOracle;
+        updateTokenPrice(_token);
+    }
+
+    function distribute(bytes memory ctx)
+        public
+        override
+        returns (bytes memory newCtx)
+    {
+        newCtx = ctx;
+
+        require(
+            market
+                .oracles[market.outputPools[OUTPUTA_INDEX].token]
+                .lastUpdatedAt >= block.timestamp - 3600,
+            "!currentValueA"
+        );
+        require(
+            market
+                .oracles[market.outputPools[OUTPUTB_INDEX].token]
+                .lastUpdatedAt >= block.timestamp - 3600,
+            "!currentValueB"
+        );
+
+        // At this point, we've got enough of tokenA and tokenB to perform the distribution
+        uint256 tokenAAmount = inputTokenA.balanceOf(address(this));
+        uint256 tokenBAmount = inputTokenB.balanceOf(address(this));
+
+        // Check how much inputTokenA we have already from tokenB
+        uint256 tokenHave = (tokenBAmount *
+            market.oracles[inputTokenB].usdPrice) /
+            market.oracles[inputTokenA].usdPrice;
+        // If we have more tokenA than we need, swap the surplus to inputTokenB
+        if (tokenHave < tokenAAmount) {
+            tokenHave = tokenAAmount - tokenHave;
+            _swap(inputTokenA, inputTokenB, tokenHave);
+            // Otherwise we have more tokenB than we need, swap the surplus to inputTokenA
+        } else {
+            tokenHave =
+                (tokenAAmount * market.oracles[inputTokenA].usdPrice) /
+                market.oracles[inputTokenB].usdPrice;
+            tokenHave = tokenBAmount - tokenHave;
+            _swap(inputTokenB, inputTokenA, tokenHave);
+        }
```

Comment:
Double check, is all this necessary? I'm not sure it is, I thought this was implemented in REXMarket too already, so you're overriding it here